plugins/modules/site_workflow_manager.py

```diff
@@ -1088,15 +1179,35 @@ def update_floor(self, site_params):
             dict: The API response from the 'updates_a_floor' operation or None if an exception occurs.
         """
         response = None
-        units_of_measure = "feet"
+        units_of_measure = ["feet", "meters"]
+        rf_model = [
+            "Free Space",
+            "Outdoor Open Space",
+            "Cubes And Walled Offices",
+            "Indoor High Ceiling",
+            "Drywall Office Only"
+        ]
         try:
             self.log("Updating floor with parameters: {0}".format(site_params), "INFO")
             parent_name = site_params.get("site", {}).get("floor", {}).get("parentName")
             parent_id = self.get_parent_id(parent_name)
             site_params['site']['floor']['parentId'] = parent_id
-            site_params['site']['floor']['unitsOfMeasure'] = units_of_measure
+
+            if config.get("site", {}).get("floor", {}).get("units_of_measure") not in units_of_measure:
```

Comment:
config.get("site", {}).get("floor", {}).get("units_of_measure") multiple times can be avoided by storing it in a variable. This makes the code cleaner and more efficient

```
units_of_measure_value = config.get("site", {}).get("floor", {}).get("units_of_measure")
if units_of_measure_value not in units_of_measure:
    error_msg = "Given Unit of Measure: {} not in allowed units: {}".format(units_of_measure_value, units_of_measure)
    self.module.fail_json(msg=error_msg)
else:
    site_params['site']['floor']['unitsOfMeasure'] = units_of_measure_value
    self.log("Set 'units of measure' to: {}".format(units_of_measure_value), "DEBUG")
```

Reply:
Modified